cpp/job/tests.cpp

```diff
@@ -37,38 +44,54 @@ int test_simple_job() {
 }
 
 
-int test_simple_job_with_exception() {
+int test_simple_job_with_exception () {
 	JobManager manager{4};
 	manager.start();
 
-	bool finished = false;
-	bool result = -1;
+	int good_jobs = 5;
+	int bad_jobs = 5;
+	std::atomic<int> finished(0);
+	std::atomic<int> errors(0);
 
-	auto job_function = []() -> int {
+	auto exception_job_function = []() -> int {
 		throw "error string";
 	};
+
+	auto job_function = []() -> int {
+		return 42;
+	};
+
 	auto job_callback = [&](result_function_t<int> get_result) {
 		try {
 			get_result();
 		} catch (const char *e) {
-			result = 0;
+			errors++;
 		}
-		finished = true;
-
+		finished++;
 	};
-	manager.enqueue<int>(job_function, job_callback);
 
-	while (not finished) {
+	for (int i = 0; i < good_jobs; i++) {
+		manager.enqueue<int>(job_function, job_callback);
+	}
```

Comment:
why did you add the non-excepting jobs here? they should be tested in the other function?